docs/jwst/skymatch/description.rst

```diff
@@ -0,0 +1,239 @@
+Description
+===========
+
+Overview
+--------
+The ``skymatch`` step can be used to compute sky values in a collection of input
+images that contain both sky and source signal. The sky values can be computed
+for each image separately or in a way that matches the sky levels amongst the
+collection of images so as to minimize their differences. This operation is
+typically applied before combining multiple images into a mosaic. When running
+the ``skymatch`` step in a matching mode, it compares *total* signal levels
+in *the overlap regions* (instead of doing this comparison on a per-pixel basis)
+of a set of input images and computes the signal offsets for each image that will
+minimize -- in a least squares sense -- the residuals across the entire set.
+This comparison is performed directly on the input images without resampling
+them onto a common grid. The overlap regions are computed directly on the sky
+(celestial sphere) for each pair of input images. By default the sky value
+computed for each image is recorded, but not actually subtracted from the
+images. Also note that the meaning of "sky background" depends on the chosen
+sky computation method.
+
+Assumptions
+-----------
+When matching sky background, the code needs to compute bounding polygon
+intersections in world coordinates. The input images, therefore, need to have
+a valid WCS, generated by the :ref:`assign_wcs <assign_wcs_step>` step.
+
+Algorithm
+---------
+The ``skymatch`` step provides several methods for constant sky background
+value computations.
+
+The first method, called "local", essentially is an enhanced version of the
+original sky subtraction method used in older versions of
+`astrodrizzle <https://drizzlepac.readthedocs.io/en/latest/astrodrizzle.html>`_.
+This method simply computes the mean/median/mode/etc. value of the "sky" separately
+in each input image. This method was upgraded to be able to use DQ flags
+to remove bad pixels from being used in computations of the sky statistics.
+
+In addition to the classic "local" method,
+two other methods have been introduced: "global" and
+"match", as well as a combination of the two -- "global+match".
+
+- The "global" method computes a single sky value across *all*
+  input images. The resulting *single sky value* is then considered to be
+  the background in *all input images*. The global value is always based on
+  the image with the lowest sky signal level.
```

Comment:
So the algorithm of global to select the image is "local" at its base, i.e. the global method uses the lowest "local" value as the global value? On first read, it wasn't clear exactly what the " is always based on" means  - perhaps just 

> The global value uses the local method on each image and selects the lowest value.

Reply:
I agree, it is not very clear. Does this mean that, if the "subtract" flag is set to True, the background that is subtracted to all images is exactly the lowest one and the differences between images are not taken in account?

Reply:
Yes, "global" by itself essentially uses the "local" method to find the background level in each image independently and then use the lowest result as the background for the entire set. That single (lowest) value is what will be used to populate the BKGLEVEL keyword in all images and what would be used to do the subtraction if that parameter were turned on. That would result in one image having (near) zero bkg, while the remaining ones would have some positive residual in them.